src/lib/ccf/index.ts

```diff
@@ -0,0 +1,458 @@
+   *  @param {string} staticParams.instance_type instance type from the list of supported instances
+   *  @param {number} staticParams.expected_lifespan expected lifespan of the instance in years
+   *  @param {Interpolation} staticParams.interpolation linear(All Clouds), spline (only for AWS)
+   */
+  async configure(
+    name: string,
+    staticParams: object | undefined = undefined
+  ): Promise<IImpactModelInterface> {
+    this.name = name;
+
+    if (staticParams === undefined) {
+      throw new Error('Required Parameters not provided');
+    }
+
+    if ('provider' in staticParams) {
+      const provider = staticParams?.provider as string;
+      if (['aws', 'gcp', 'azure'].includes(provider)) {
+        this.provider = provider;
+      } else {
+        throw new Error('Provider not supported');
+      }
+    } else {
+      throw new Error('Provider not provided');
+    }
+
+    if ('instance_type' in staticParams) {
+      const instanceType = staticParams?.instance_type as string;
+      if (instanceType in this.computeInstances[this.provider]) {
+        this.instanceType = instanceType;
+      } else {
+        throw new Error('Instance Type not supported');
+      }
+    } else {
+      throw new Error('Instance Type not provided');
+    }
+
+    if ('expected_lifespan' in staticParams) {
+      this.expectedLifespan = staticParams?.expected_lifespan as number;
+    }
+
+    if ('interpolation' in staticParams) {
+      if (this.provider !== 'aws') {
+        throw new Error('Interpolation method not supported');
+      }
+      const interpolation = staticParams?.interpolation as Interpolation;
+      if (Object.values(Interpolation).includes(interpolation)) {
+        this.interpolation = interpolation;
+      } else {
+        throw new Error('Interpolation method not supported');
+      }
+    }
+
+    return this;
+  }
+
+  /**
+   * Calculate the total emissions for a list of observations
+   *
+   * Each Observation require:
+   *  @param {Object[]} observations  ISO 8601 datetime string
+   *  @param {string} observations[].datetime ISO 8601 datetime string
+   *  @param {number} observations[].duration observation duration in seconds
+   *  @param {number} observations[].cpu percentage cpu usage
+   */
+  async calculate(
+    observations: object | object[] | undefined
+  ): Promise<object> {
+    if (observations === undefined) {
+      throw new Error('Required Parameters not provided');
+    }
+
+    if (this.instanceType === '' || this.provider === '') {
+      throw new Error('Configuration is incomplete');
+    }
+
+    const results: ICcfResult[] = [];
+    if (Array.isArray(observations)) {
+      observations.forEach((observation: KeyValuePair) => {
+        const e = this.calculateEnergy(observation);
+        const m = this.embodiedEmissions(observation);
+
+        results.push({
+          e: e,
+          m: m,
+        });
+      });
+    }
```

Comment:
instead of declaring empty array then collecting, we can reduce obsercations array

```
return observations.reduce((observation) => {
  const calculatedObservation = {
    e: this.calculateEnergy(observation)
    m: this.embodiedEmissions(observation)
  }
  
  acc.push(calculatedObservation)
  
  return acc
}, [])
```